Sources/NeedleFoundation/Internal/DependencyProviderRegistry.swift

```diff
@@ -68,8 +81,22 @@ public class __DependencyProviderRegistry {
         }
     }
 
+    /// Retrieve the dependency provider for the given componentpath.
+    ///
+    /// - parameter componentpath: The component path that uses the returned dependency provider.
+    /// - returns: The dependency provider for the given componentpath.
+    public func dependencyProviderFactory(`for` componentPath: String) -> ((Scope) -> AnyObject)? {
+        providerFactoryLock.lock()
+        defer {
+            providerFactoryLock.unlock()
+        }
+        
+        return providerFactories[componentPath.hashValue]
+    }
+
     private let providerFactoryLock = NSRecursiveLock()
-    private var providerFactories = [Int: (Scope) -> AnyObject]()
+    // marked internal to allow testing
+    var providerFactories = [Int: (Scope) -> AnyObject]()
```

Comment:
This is pretty dangerous right? If people do `@testable import`, this will become visible as well.

Reply:
Updated this and modified the test code to rely on an alternate mechanism.